detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/Detekt.kt

```
@@ -240,10 +257,29 @@ open class Detekt : SourceTask(), VerificationTask {
                     if (task is Detekt) task.xmlReportFile.orNull?.asFile else null
                 }
             }
-            if (!xmlReports.isEmpty() && debugOrDefault) {
+            if (xmlReports.isNotEmpty() && debugOrDefault) {
                 logger.info("Merging report files of subprojects $xmlReports into $xmlReportTargetFileOrNull")
             }
             mergeXmlReports(xmlReportTargetFileOrNull.asFile, xmlReports)
         }
     }
+
+    @Suppress("TooGenericExceptionCaught")
+    private fun Iterable<File>.getKotlinLanguageVersion(): String? {
+        val urls = map { it.toURI().toURL() }
+        if (urls.isNotEmpty()) {
+            URLClassLoader(urls.toTypedArray()).use { classLoader ->
+                try {
+                    val clazz = classLoader.loadClass("kotlin.KotlinVersion")
+                    val field = clazz.getField("CURRENT")
+                    field.isAccessible = true
+                    val versionObj = field.get(null)
+                    return versionObj?.toString()
+                } catch (e: Throwable) {
```

Are you sure that we need to catch all exceptions?
Are there any specific exceptions which could occur in the try-catch block?

---

Yes, we can try to catch specific exceptions. But here my goal was to remain the code as stable as possible. If you don't mind, I'd like to leave as it is.

---

Accepted